pkg/monitor/monitor.go

```diff
@@ -91,16 +93,20 @@ func (mon *monitor) Run(ctx context.Context) error {
 		err = mon.listBuckets(ctx)
 		if err != nil {
 			mon.baseLog.Error(err)
+		} else {
+			mon.lastBucketlist = time.Now()
 		}
 
 		<-t.C
 	}
 }
 
-// checkReady checks the ready status of the frontend to make it consistent
+// checkReady checks the ready status of the monitor to make it consistent
 // across the /healthz/ready endpoint and emitted metrics.   We wait for 2
 // minutes before indicating health.  This ensures that there will be a gap in
 // our health metric if we crash or restart.
 func (mon *monitor) checkReady() bool {
-	return time.Now().Sub(mon.startTime) > 2*time.Minute
+	return (time.Now().Sub(mon.lastBucketlist) < time.Minute) &&
```

Comment:
@Makdaam  can you please leave some comment on the logic here. Currently, we explain `now - startTime >2min` but `now - lastBucketlis < 1min` is not quite clear to me what was the thinking behind this complex `cond AND cond AND condAlwaysTrueAfter2Min`.

Reply:
added comments explaining these lines.